atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java

```diff
@@ -86,12 +142,20 @@ public InMemoryKeyValueService(boolean createTablesAutomatically) {
         this(createTablesAutomatically, MoreExecutors.newDirectExecutorService());
     }
 
-    public InMemoryKeyValueService(boolean createTablesAutomatically,
-                                   ExecutorService executor) {
+    public InMemoryKeyValueService(boolean createTablesAutomatically, ExecutorService executor) {
         super(executor);
         this.createTablesAutomatically = createTablesAutomatically;
     }
 
+    /**
+     * Creates an InMemoryKvs that can be asynchronously initialized. If initializeAsync is true, the returned Kvs will
+     * initially be uninitialized, and will only succeed at the second attempt at initialization.
+     */
+    public static KeyValueService create(boolean createTablesAutomatically, boolean initializeAsync) {
```

Comment:
we shouldn't need this... The InMemoryKvs should never need initializing

Reply:
it's a bunch of hashmaps

Reply:
if we want to have behaviour like this in our tests, we should just have a test kvs which does this; the InMemoryKvs is closer to prod code and so shouldn't really contain extraneous stuff